src/main.py

```diff
@@ -0,0 +1,56 @@
+# OMIT !/usr/bin/env python3
+from pathlib import Path
+import click
+from io import StringIO
+
+from src.option_select import select
+from src.run_cocci import run_cocci
+
+@click.command
+@click.option(
+    "-i",
+    "--input",
+    type=Path,
+    required=True,
+    help="Directory containing your C source files.",
+)
+@click.option(
+    '--scope',
+    type=click.Choice(['all', 'select']), 
+    prompt='Choose an option to compare all or just a single file'
+)
+@click.option(
+    "-v",
+    "--verbosity", 
+    type=str,
+    default=0,
+    required=False,
+    help = "Enter 0 or 1"
+)
+def cli(input, scope, verbosity):
+    if not input.is_dir():
+        raise Exception(f"input path is not valid")
+
+    patch_list = (Path.cwd() / "cocci").glob("*.cocci")
+    patch_list = list(map(lambda file: file.name, patch_list))
+
+    if(scope=='select'):
+        patch_list = [select(patch_list)]
+
+    input_list = []
+    pattern = ['*.c','*.cpp']
```

Comment:
Make sure that this works recursively, if you have a directory inside a directory (that's pretty common)